java/fury-core/src/main/java/org/apache/fury/serializer/ArraySerializers.java

```diff
@@ -85,52 +103,69 @@ public void write(MemoryBuffer buffer, T[] arr) {
       int len = arr.length;
       RefResolver refResolver = fury.getRefResolver();
       Serializer componentSerializer = this.componentTypeSerializer;
-      int header = componentSerializer != null ? 0b1 : 0b0;
-      buffer.writeVarUint32Small7(len << 1 | header);
-      if (componentSerializer != null) {
-        for (T t : arr) {
-          if (!refResolver.writeRefOrNull(buffer, t)) {
-            componentSerializer.write(buffer, t);
-          }
-        }
+      if (this.collectionSerializer != null) {
+        fury.getGenerics().pushGenericType(this.collectionGenericType);
+        ArrayAsList list = new ArrayAsList(0);
```

Comment:
Could we cache this object as a field.  Nested serialization is a rare case, if we cache it as a field, we could save object creation cost. 

We could write like :
```java
ArrayAsList list = this.list;
if (list == null) {
  list = new ArrayAsList(0);
} else {
  this.list = null
}

// before serialiazation

...


// after serialization

this.list = list

```